server/src/main/antlr4/org/eclipse/lsp/cobol/core/parser/CommonCobolLexer.g4

```diff
@@ -207,145 +200,54 @@ VOLUME : V O L U M E;
-SEMICOLON_FS : ';' ('\r' | '\n' | '\f' | '\t' | ' ')+ | ';' EOF;
-SINGLEQUOTE : '\'';
-RPARENCHAR : ')';
 SEPARATE : S E P A R A T E;
 SEQUENCE : S E Q U E N C E;
 SIZE : S I Z E;
-SLASHCHAR : '/';
 SOURCE : S O U R C E;
 SPACE : S P A C E;
-SQLLINECOMMENTCHAR: '--';
 SUPPRESS : S U P P R E S S;
 TERMINAL : T E R M I N A L;
 TEST : T E S T;
 THREAD : T H R E A D;
 UPPER : U P P E R;
 
-LEVEL_NUMBER : ([1-9])|([0][1-9])|([1234][0-9]);
-LEVEL_NUMBER_66 : '66';
-LEVEL_NUMBER_77 : '77';
-LEVEL_NUMBER_88 : '88';
-
-INTEGERLITERAL : (PLUSCHAR | MINUSCHAR)? DIGIT+ | LEVEL_NUMBER;
-
-// DECIMAL_CONST : DIGIT DIGIT DIGIT DIGIT DOT DIGIT {sqlFlag}? ;
-SINGLEDIGITLITERAL : DIGIT {sqlFlag}? ;
-
-NUMERICLITERAL : (PLUSCHAR | MINUSCHAR)? DIGIT* (DOT | COMMACHAR) DIGIT+ (('e' | 'E') (PLUSCHAR | MINUSCHAR)? DIGIT+)?;
-
-NONNUMERICLITERAL : UNTRMSTRINGLITERAL | STRINGLITERAL | DBCSLITERAL | HEXNUMBER | NULLTERMINATED;
-
-//TXTLITERAL : STRINGLITERAL | IDENTIFIER;
-CHAR_STRING_CONSTANT : HEXNUMBER | STRINGLITERAL;
-
-IDENTIFIER : ([a-zA-Z0-9]+ ([-_]+ [a-zA-Z0-9]+)*);
-COPYBOOK_IDENTIFIER : [a-zA-Z0-9#@$]+ ([-_]+ [a-zA-Z0-9#@$]+)*;
-FILENAME : IDENTIFIER+ '.' IDENTIFIER+;
-
-OCTDIGITS : OCT_DIGIT {sqlFlag}? ;
-HEX_NUMBERS : HEXNUMBER {sqlFlag}? ;
-
-// whitespace, line breaks, comments, ...
-NEWLINE : '\r'? '\n' -> channel(HIDDEN);
-COMMENTLINE : COMMENTTAG WS ~('\n' | '\r')* -> channel(HIDDEN);
-COMMENTENTRYLINE : COMMENTENTRYTAG WS ~('\n' | '\r')*  -> channel(HIDDEN);
-WS : [ \t\f;]+ -> channel(HIDDEN);
-SEPARATOR : ', ' {!sqlFlag}? -> channel(HIDDEN);
-
-//SQL comments
-SQLLINECOMMENT
-	:	SQLLINECOMMENTCHAR ~[\r\n]* NEWLINE {sqlFlag}? -> channel(HIDDEN)
-	;
-
-// treat all the non-processed tokens as errors
-ERRORCHAR : . ;
-
-ZERO_DIGIT: '0';
-
-
-fragment HEXNUMBER :
-	X '"' [0-9A-F]+ '"'
-	| X '\'' [0-9A-F]+ '\''
-;
-
-fragment NULLTERMINATED :
-	Z '"' (~["\n\r] | '""' | '\'')* '"'
-	| Z '\'' (~['\n\r] | '\'\'' | '"')* '\''
-;
-
-fragment STRINGLITERAL :
-	'"' (~["\n\r] | '""' | '\'')* '"'
-	| '\'' (~['\n\r] | '\'\'' | '"')* '\''
-;
-
-fragment UNTRMSTRINGLITERAL :
-	'"' (~["\n\r] | '""' | '\'')*
-	| '\'' (~['\n\r] | '\'\'' | '"')*
-;
-
-fragment DBCSLITERAL :
-	[GN] '"' (~["\n\r] | '""' | '\'')* '"'
-	| [GN] '\'' (~['\n\r] | '\'\'' | '"')* '\''
-;
-
-fragment
-OCT_DIGIT        : [0-8] ;
-fragment DIGIT: OCT_DIGIT | [9];
-// case insensitive chars
-fragment A:('a'|'A');
-fragment B:('b'|'B');
-fragment C:('c'|'C');
-fragment D:('d'|'D');
-fragment E:('e'|'E');
-fragment F:('f'|'F');
-fragment G:('g'|'G');
-fragment H:('h'|'H');
-fragment I:('i'|'I');
-fragment J:('j'|'J');
-fragment K:('k'|'K');
-fragment L:('l'|'L');
-fragment M:('m'|'M');
-fragment N:('n'|'N');
-fragment O:('o'|'O');
-fragment P:('p'|'P');
-fragment Q:('q'|'Q');
-fragment R:('r'|'R');
-fragment S:('s'|'S');
-fragment T:('t'|'T');
-fragment U:('u'|'U');
-fragment V:('v'|'V');
-fragment W:('w'|'W');
-fragment X:('x'|'X');
-fragment Y:('y'|'Y');
-fragment Z:('z'|'Z');
+C_CHAR: C {!sqlFlag}?;
```

Comment:
This looks alarming...

Reply:
Agree, it looks terrible. Separating the parser should make it easier